### PR TITLE
types for dta library were added

### PR DESCRIPTION
**Pull request by MakarSantalov**

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
+export type DtaEngines = 'dynamodb' | 'neo4j';
```
Comment (paulcruse-syn) [inline types/index.d.ts]: This needs to be broken out into different files to make this more extensible; basically 1 file per adapter type.

This file is too large.

**Review by paulcruse-syn — CHANGES_REQUESTED** [review]

**Comment by syngenta-aleksei-zolotykh** [comment]
@MakarSantalov it looks like we need to add type property to package.json files

See documentation [here](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package)

File: typings/common.d.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+export declare type BaseAdapterParams<TEngine extends string>  = {
```
Comment (paulcruse-syn) [inline typings/common.d.ts]: Common, similar to utils, is considered an anti-pattern. 

Please break these up into separate files

File: typings/neo4j.d.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import {
```
Comment (paulcruse-syn) [inline typings/neo4j.d.ts]: This is too much in 1 file, please break these up into separate files.

File: typings/dynamodb.d.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
+import {
+    DataParams,
+    QueryParams,
+    BaseAdapterParams
+} from './common';
+
+declare type DynamoDBQueryOperation = 'query';
```
Comment (paulcruse-syn) [inline typings/dynamodb.d.ts]: This is too much in 1 file, please break these up into separate files.

**Review by paulcruse-syn — CHANGES_REQUESTED** [review]

**Review by paulcruse-syn — APPROVED** [review]

**Pull request merged**